### PR TITLE
Added durable state current persistence ids query

### LINKED ISSUE #30795: Durable state store persistence ids query
It would be good to have a query on durable state stores that returned all persistence ids, for back-office/maintenance type tasks, similar to `CurrentPersistenceIdsQuery`. As with #30794, ideally this would be pageable/resumable.

### PR DESCRIPTION
**Pull request by jroper**
Fixes #30795

Adds a query for durable state objects to get all persistence IDs.

I only provided the current persistence ids query - a continuous stream would be much harder to implement, since it would require tracking all persistence ids to know if one is new.

This query does provide paging, using the persistence offset abstraction. This allows databases to implement it in a way that makes sense for them, for many, the offset will be the persistence id itself, since the persistence id is the primary key and so the index will be sorted by persistence id, and this will allow the database to jump straight to the right location in the index.

I also included a limit parameter - not strictly necessary as long as the database is streaming returned results, but will perform a little better even when streaming as it will eliminate buffered and in flight results when the stream is cancelled from being loaded at all.

### CODE REVIEW

**Review by patriknw — APPROVED** [review]
LGTM, with a small naming suggestion or question

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-persistence-query/src/main/scala/akka/persistence/query/scaladsl/CurrentDurableStatePersistenceIdsQuery.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+   * Not all durable state plugins may support in database paging, and may simply use drop/take Akka streams operators
+   * to manipulate the result set according to the paging parameters.
+   *
+   * @param afterId The ID to start returning results from, or [[None]] to return all ids. This should be an id
+   *                returned from a previous invocation of this command. Callers should not assume that ids are
+   *                returned in sorted order.
+   * @param limit The maximum results to return. Use Long.MaxValue to return all results. Must be greater than zero.
+   * @return A source containing all the persistence ids, limited as specified.
+   */
+  def currentPersistenceIds(afterId: Option[String], limit: Long): Source[String, NotUsed]
```
Comment (patriknw) [inline akka-persistence-query/src/main/scala/akka/persistence/query/scaladsl/CurrentDurableStatePersistenceIdsQuery.scala:28]: @jroper was there a reason for why we are not adding the same for event sourced?
maybe @raboof already asked this and I just missed the answer?

Comment (jroper) [reply akka-persistence-query/src/main/scala/akka/persistence/query/scaladsl/CurrentDurableStatePersistenceIdsQuery.scala:28]: No reason other than I thought they were different enough to warrant being done in separate PRs.

**Pull request merged**